spidermon/contrib/stats/statscollectors/local_storage.py

```diff
@@ -9,7 +9,8 @@
 class LocalStorageStatsHistoryCollector(StatsCollector):
     def _stats_location(self, spider):
         statsdir = data_path("stats", createdir=True)
-        return os.path.join(statsdir, f"{spider.name}_stats_history")
+        spider_name = os.getenv("SHUB_VIRTUAL_SPIDER", spider.name)
```

Comment:
Refactor spider name extraction into a utility function that can be reused across all methods. 
That way we can introduce a setting to change it for a different one if wanted.